pkg/components/exec_node_base.go

```diff
@@ -3,56 +3,147 @@ package components
 import (
 	"context"
 	"log"
+	"path"
 
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/yaml"
 	ptr "k8s.io/utils/pointer"
 
+	ytv1 "github.com/ytsaurus/yt-k8s-operator/api/v1"
+
+	"github.com/ytsaurus/yt-k8s-operator/pkg/consts"
 	"github.com/ytsaurus/yt-k8s-operator/pkg/resources"
 	"github.com/ytsaurus/yt-k8s-operator/pkg/ytconfig"
 )
 
 type baseExecNode struct {
-	server     server
-	cfgen      *ytconfig.NodeGenerator
-	sidecars   []string
-	privileged bool
+	server server
+	cfgen  *ytconfig.NodeGenerator
+	spec   *ytv1.ExecNodesSpec
+
+	sidecarConfig *ConfigHelper
+}
+
+func (n *baseExecNode) IsJobResourcesIsolated() bool {
+	if n.spec.JobEnvironment != nil && n.spec.JobEnvironment.CRI != nil {
```

Comment:
Just `return n.spec.JobEnvironment != nil && n.spec.JobEnvironment.CRI != nil`?

Reply:
Now this piece is more complicated.